status/checkMissingPods.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 gRPC authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+	"github.com/grpc/test-infra/config"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// LoadTestMissing defines missing pods of LoadTest.
+type LoadTestMissing struct {
+	// Driver is the component that orchestrates the test. If Driver is not set
+	// that means we already have the Driver running.
+	Driver *grpcv1.Driver `json:"driver,omitempty"`
+
+	// Servers are a list of components that receive traffic from. The list
+	// indicates the Servers still in need.
+	Servers []grpcv1.Server `json:"servers,omitempty"`
+
+	// Clients are a list of components that send traffic to servers. The list
+	// indicates the Clients still in need.
+	Clients []grpcv1.Client `json:"clients,omitempty"`
+}
+
+// CheckMissingPods attempts to check if any required component is missing from
+// the current load test. It takes reference of the current load test and a pod
+// list that contains all running pods at the moment, returning all missing
+// components required from the current load test with their roles.
+func CheckMissingPods(currentLoadTest *grpcv1.LoadTest, relevantPods []*corev1.Pod) *LoadTestMissing {
```

Comment:
nit: would you be willing to rename `relevantPods` to `ownedPods` for consistency?

Reply:
This is not super important.

Reply:
My bad, I change that in the controller_loadtest, forgot to change it in the separate file, will change it. :)